hoodie-common/src/main/java/com/uber/hoodie/common/table/view/HoodieTableFileSystemView.java

```diff
@@ -79,10 +88,33 @@ public HoodieTableFileSystemView(HoodieTableMetaClient metaClient,
     this.visibleActiveTimeline = visibleActiveTimeline;
     this.fileGroupMap = new HashMap<>();
     this.partitionToFileGroupsMap = new HashMap<>();
-    //TODO: vb Will be implemented in next PR
-    this.fileIdToPendingCompactionInstantTime = new HashMap<>();
-  }
 
+    // Build fileId to Pending Compaction Instants
+    List<HoodieInstant> pendingCompactionInstants =
+        metaClient.getActiveTimeline().filterPendingCompactionTimeline().getInstants().collect(Collectors.toList());
+    Map<String, String> fileIdToPendingCompactionMap = new HashMap<>();
+    pendingCompactionInstants.stream().flatMap(instant -> {
+      try {
+        HoodieCompactionWorkload workload = AvroUtils.deserializeHoodieCompactionWorkload(
```

Comment:
Rename to AvroUtils.convertCompactionWorkload() or AvroUtils.deserializeCompactionWorkload() ? Naming seems too long.